### PR TITLE
Devin -> Merge

### PR DESCRIPTION
**Pull request by b1gd3vd0g**
Fixed debug mode to work with ZONES instead of DIMENSIONS

### CODE REVIEW

File: js/classes/game_engine.js [hunk 1]
```diff
@@ -203,35 +203,74 @@ class GameEngine {
...
-                CTX.lineTo(x - CAMERA.pos.x, DIMENSION.MAX_Y - CAMERA.pos.y);
-                CTX.stroke();
-            }
+
+        // Draw all the vertical lines by iterating through the x values we need.
+        for (let x = ZONE.MIN_PT.x; x <= ZONE.MAX_PT.x; x += Block.SCALED_SIZE) {
+            // Start a path
+            CTX.beginPath();
+            // Starting point
+            CTX.moveTo(x - CAMERA.pos.x, ZONE.MIN_PT.y - CAMERA.pos.y);
```
Comment (calebkrauter) [inline js/classes/game_engine.js:222]: I'm a little confused on what moveTo does with the x and y position of camera. Could you explain this?

File: js/classes/game_engine.js [hunk 2]
```diff
@@ -203,35 +203,74 @@ class GameEngine {
...
+            // Ending point
+            CTX.lineTo(x - CAMERA.pos.x, ZONE.MAX_PT.y - CAMERA.pos.y);
```
Comment (calebkrauter) [inline js/classes/game_engine.js:224]: This makes sense to me.

File: js/classes/game_engine.js [hunk 3]
```diff
@@ -203,35 +203,74 @@ class GameEngine {
...
+        // This is the minimum y value that we will write at. Note: the +18 is related to the font size.
+        // Apparently drawing text with the context will start writing right ABOVE where you dictate.
+        const minY = ZONE.MIN_PT.y + 18;
+        // This is going to be where we want to start drawing the label, just like gameX.
         let gameY = minY;
-        for (let blockX = DIMENSION.MIN_BLOCK_X; blockX <= DIMENSION.MAX_BLOCK_X; blockX += 5) {
-            for (let blockY = DIMENSION.MIN_BLOCK_Y; blockY <= DIMENSION.MAX_BLOCK_Y; blockY += 5) {
+
+        // As it is currently working, it will DEFINITELY label ZONE.MIN_BLOCK, and every 5 other than that.
+        // Would like to refactor it eventually to DEFINITELY label the origin, but not super important.
```
Comment (calebkrauter) [inline js/classes/game_engine.js:260]: Sounds good! Could you make a TODO in the comment and link an issue for this?

File: js/classes/game_engine.js [hunk 4]
```diff
@@ -203,35 +203,74 @@ class GameEngine {
...
+        // Also, could be refactored to better use the Vector class's static methods.
+        for (let blockX = ZONE.MIN_BLOCK.x; blockX <= ZONE.MAX_BLOCK.x; blockX += 5) {
+            for (let blockY = ZONE.MIN_BLOCK.y; blockY <= ZONE.MAX_BLOCK.y; blockY += 5) {
+                // The string representing the cell we are labeling.
                 let pt = "(" + blockX + ", " + blockY + ")";
+                // Draw it.
                 CTX.fillText(pt, gameX - CAMERA.pos.x, gameY - CAMERA.pos.y, Block.SCALED_SIZE);
```
Comment (calebkrauter) [inline js/classes/game_engine.js:267]: The variables here are confusing to me. I know you mentioned refactoring, I'd agree with that when you get time.

**Comment by calebkrauter** [comment]
Looks good, just could use refactoring as you suggest. Thanks for making this!

**Review by calebkrauter — APPROVED** [review]

**Review by tclaar — APPROVED** [review]
It would probably be better to only draw the grid and labels within the camera's field of view as opposed to the entire zone. That said, good work fixing it!

**Pull request merged**